### PR TITLE
feat: household credits — Net Contribution math and Owed-to-Members KPI (#316)

### PR DESCRIPTION
**Pull request by nathanjohnpayne**
Closes #316. One vertical slice of the off-cycle-adjustments epic (#314).

Authoring-Agent: claude

## What this builds

Makes an overpaid household's **Credit** visible and correctly computed end-to-end, with **no new mutations** — a read/display tracer.

- Computes a household's **Net Contribution** (gross paid − recorded refunds/carried-forward credits) and derives its **Credit** (net amount overpaid) at the **household grain** (ADR 0001 — internal member imbalance nets out and is invisible).
- Surfaces the credit on the **settlement board**: a "Credit" box on the household card, and the card status reads from Net Contribution.
- Adds an **"Owed to Members"** dashboard KPI = sum of unresolved household credits, distinct from "Outstanding".
- Treats sub-cent rounding credits as zero via `CREDIT_EPSILON`.

## ⚠️ High-risk zone — mandatory human review

This PR modifies **`src/lib/calculations.js` → `calculateSettlementMetrics()`**, a high-risk settlement-math zone per `rules/repo_rules.md`. It also meets the Phase 4 external-review threshold (379 lines changed > 300, touches `src/lib/**`).

- `calculateSettlementMetrics` now computes each household's figures through a new pure `getHouseholdFinancials()` helper and gains a `totalCreditsOwed` field.
- With an empty `creditAdjustments` array (today's state — no refunds exist yet), `totalAnnual`, `totalPayments`, `totalOutstanding`, and `percentage` are **unchanged**. `paidCount` uses `netContribution >= owed − CREDIT_EPSILON`, equivalent to the prior `payment >= combinedTotal` for every non-sub-cent amount (see observation 2).

## ADR alignment

- **ADR 0001 (household grain):** credit/net-contribution/settled are summed across primary + linked members before differencing, so internal imbalance (one over, one under) nets to zero household credit. Tested.
- **"Refunded household reads Settled":** the board status derives from Net Contribution, so gross-paid > owed but net == owed (after a refund) reads **Settled**, not Overpaid. Tested.
- **ADR 0006 (credits don't block close):** credits are display-only here; an overpaid household still counts as settled. No gate change.
- **No mutations:** `creditAdjustments[]` is only *read*; `DashboardView` sources it with a `[]` default. Live wiring into service state lands with the mutation slices (#318/#321/#322). The subtraction path is proven by tests with synthetic refund/carry-forward records.

## Scope boundaries (deliberately NOT here)

- No refund / carry-forward / usage-charge **mutations** (later slices).
- `src/lib/BillingYearService.js` **untouched** (no new ledger state).
- Per-member "Overpaid" badges on linked rows stay gross-based (display breakdown only, ADR 0001).

## Self-Review

**Correctness.** `getHouseholdFinancials` sums owed/paid/adjustments across the household before differencing; epsilon zeroes sub-cent residue; `getPaymentStatus` is fed (epsilon-snapped) Net Contribution so the refunded case reads Settled. Tests cover overpayment, refund-to-settled, carry-forward, internal-imbalance netting, and epsilon, plus the Credit box, Overpaid→Settled, and the KPI.

**Regression risk.** `calculateSettlementMetrics` is consumed by React and the legacy app (3-arg wrapper). The 4th param is optional; with no adjustments outputs are byte-identical except the `paidCount` boundary now tolerates sub-cent underpayment (observation 2). Full suite green: **663 React + 288 legacy**, secret scan clean. The Balance box becomes a Credit box only when `credit > 0`; gross Paid/Balance, Record-Payment gating, and email-invoice "settled" logic are unchanged.

**Style.** New helpers mirror `getPaymentTotalForMember` and the file's pure-function convention. CSS reuses the documented "credit = green" three-state model already in `shell.css`.

**Test coverage.** +23 tests (calculations, SettlementBoard, DashboardView). Specs updated alongside (`billing-calculations.md`, `settlement-board.md`); `check_spec_test_alignment` passes.

**Security / dependency hygiene.** No new deps. No `firestore.rules` / `storage.rules` / Cloud Functions changes (no mutations, no new data paths). Secret scan passes. App *records*, doesn't move money — unchanged.

## Observations for the reviewer (not blockers)

1. **Global `totalOutstanding` nets overpayments across households** — `max(0, totalAnnual − totalPayments)` is a global sum, so one household's overpayment lowers the headline outstanding figure. Pre-existing; preserved here, out of scope for #316.
2. **`paidCount` epsilon tolerance** — the settled boundary now treats sub-cent *under*payment as settled (symmetric with the credit epsilon). Arguably a latent fix; called out since it lives in the high-risk function.

## Test plan

- `npm test` → legacy (288) + React (663) + secret scan: green locally.
- repo_lint `check_*` scripts: all pass.

### CODE REVIEW

**Review by nathanpayne-claude — COMMENTED** [review]
## Internal reviewer pass — nathanpayne-claude

Reviewing as the internal reviewer identity. This is an **above-threshold** PR (379 > 300) touching a **high-risk** zone (`src/lib/calculations.js`), so I post `--comment` only; the cross-agent gate is carried by Phase 4, and the settlement-math change additionally requires **explicit human sign-off** before merge (`rules/repo_rules.md` § High-Risk Modification Zones).

### High-risk settlement math — verified
- `getHouseholdFinancials` sums owed / gross paid / credit adjustments across the primary + linked members **before** differencing, so internal member imbalance nets out (ADR 0001). Confirmed by the "internal imbalance" unit + component tests.
- `calculateSettlementMetrics`: for an empty `creditAdjustments` array (the only state that exists today), `totalAnnual`, `totalPayments`, `totalOutstanding`, and `percentage` are unchanged vs `main`. Verified against the existing React (`calculations.test.js`) and legacy (`billing.test.js`, `ffb-mitigation-plan.test.js`) suites — both green.
- Credit is epsilon-zeroed **per household before summation**, so `totalCreditsOwed` does not accumulate sub-cent dust across households.

### Display
- Settlement-board household status now reads from Net Contribution (epsilon-snapped), so a refunded household (gross > owed, net == owed) reads **Settled**, not Overpaid; the Credit box appears only when `credit > 0` (which implies gross balance <= 0, so it never collides with a positive Balance). Verified.
- "Owed to Members" KPI is distinct from "Outstanding".

### Serialization review (per operating-rules.md)
No serialization layer is introduced — `creditAdjustments[]` is read as structured data; no flatten/parse round-trip. N/A.

### Two items routed to the external reviewer + human
1. **Global `totalOutstanding` nets overpayments across households** (pre-existing `max(0, totalAnnual − totalPayments)`). Preserved here, out of scope for #316, but worth confirming as the credit model matures.
2. **`paidCount` settled boundary now tolerates sub-cent underpayment** (`>= owed − CREDIT_EPSILON`), symmetric with the credit epsilon. Equivalent to prior behavior for all non-sub-cent amounts; flagged because it lives in the high-risk function.

### Verdict
Internally LGTM. Not approving from this identity (above-threshold + same authoring agent). Routing to Phase 4 external review and to the human for the high-risk merge decision.

File: tests/react/views/DashboardView.test.jsx [hunk 1]
```diff
@@ -172,4 +172,25 @@ describe('DashboardView', () => {
...
+        renderDashboard({
+            payments: [
+                { memberId: 1, amount: 600, method: 'cash', note: '', date: new Date().toISOString() },
+                { memberId: 2, amount: 668.98, method: 'cash', note: '', date: new Date().toISOString() }
+            ]
+        });
+        expect(screen.getByText('Owed to Members')).toBeInTheDocument();
+        // Appears on the KPI card (also mirrored on Bob's settlement-board credit box)
+        expect(screen.getAllByText('$68.98').length).toBeGreaterThanOrEqual(1);
+    });
```
[the remark this answers is not in the document]
Comment (nathanjohnpayne) [reply tests/react/views/DashboardView.test.jsx:205]: Addressed in 4d8d76b. Both "Owed to Members" KPI tests now scope their assertions to the `.kpi-card` container (`within` / `closest`) so they verify the dashboard KPI path directly, rather than matching the same text elsewhere on the page — the credit is intentionally mirrored on Bob's settlement-board card, and "Outstanding" also renders as settlement-board status badges. Full React suite green (663).

**Review by nathanpayne-codex — CHANGES_REQUESTED** [review]
## Changes requested

I found one blocking correctness issue on the current head.

### P1 - Refunded/carry-forward households can be net-underpaid while the board still shows gross-paid balance and hides payment collection

`SettlementBoard` now computes household status from Net Contribution (`netContribution`, lines 73-76), which matches the domain model. But the actionable balance is still computed from gross payments (`const balance = combinedTotal - payment`, line 74) and reused for the payment CTA (`showPaymentAction`, line 209), the Balance/Paid header (line 289), and the payment dialog balance (lines 439-441).

That splits the board into two different truths when a recorded refund or carry-forward reduces a household below its owed amount. Example:

- household owed: `$600.00`
- gross paid: `$668.98`
- recorded refund/carry-forward: `$100.00`
- net contribution: `$568.98`

The status correctly becomes Outstanding/Partial because net contribution is below owed, but `balance = 600 - 668.98` is negative, so the header says `Paid` and `Record Payment` is hidden. The household is visibly unsettled with no obvious path to collect the remaining net shortfall.

Please derive the action/display balance from the same Net Contribution used for status, e.g. `netBalance = combinedTotal - netContribution`, and use that for status-adjacent balance text, the Record Payment gate, and the payment dialog. If the intended invariant is that `creditAdjustments` can never reduce Net Contribution below owed, this PR needs an explicit invariant test and enforcement/documentation at the mutation/import boundary; otherwise later slices can create this broken UI state.

File: src/lib/calculations.js [hunk 2]
```diff
@@ -97,30 +119,68 @@ export function getParentMember(familyMembers, memberId) {
...
-        });
-        const payment = getPaymentTotalForMember(payments, member.id) +
-            (member.linkedMembers || []).reduce((s, id) => s + getPaymentTotalForMember(payments, id), 0);
-        totalAnnual += combinedTotal;
-        totalPayments += payment;
-        if (combinedTotal <= 0 || payment >= combinedTotal) paidCount++;
+        const { owed, grossPaid, netContribution, credit } =
+            getHouseholdFinancials(member, summary, payments, creditAdjustments);
+        totalAnnual += owed;
+        totalPayments += grossPaid;
```
[the remark this answers is not in the document]
Comment (nathanjohnpayne) [reply src/lib/calculations.js:186]: Fixed in 0843b96. `totalOutstanding` is now the sum of per-household net shortfalls (`max(0, owed − netContribution)`) and `percentage` derives from it, so the refunded $68.98 no longer offsets Bob's debt — this example now reports $100, not $31.02. Covered by new tests: "does not let a recorded refund mask another household shortfall" and "does not let an overpayment mask another household shortfall or count as progress".

**Review by nathanpayne-codex — CHANGES_REQUESTED** [review]
## Changes requested

The original P1 board issue is fixed on `0843b96`: status, Balance, the Record Payment gate, the payment dialog, and the metrics now use Net Contribution consistently, and the new tests cover the refunded/over-disposed cases. Local `npm test` passed for me as well (288 legacy, 666 React, secret scan).

I found one remaining blocking wiring issue on the current head:

### P1 - `creditAdjustments` are consumed by the UI but are not loaded or preserved by the service state

`DashboardView` now reads `creditAdjustments` from `useBillingData()` and passes it into both `calculateSettlementMetrics()` and `SettlementBoard` (`src/app/views/Dashboard/DashboardView.jsx:22`, `:53`, `:182`). But the service/persistence layer never carries that field:

- `BillingYearService` initial state and unauthenticated reset omit `creditAdjustments` (`src/lib/BillingYearService.js:84-93`, `:156-160`).
- `_loadYearData()` sets `activeYear`, `familyMembers`, `bills`, `payments`, `billingEvents`, and `settings`, but not `creditAdjustments` (`src/lib/BillingYearService.js:244-250`).
- `normalizeYearData()` returns no `creditAdjustments`, so raw Firestore year data drops the array before React can see it (`src/lib/persistence.js:77-86`).
- `buildSavePayload()` also omits the field (`src/lib/persistence.js:31-40`), and `save()` writes with `setDoc()` rather than merge (`src/lib/BillingYearService.js:292-297`), so any existing stored `creditAdjustments` would be deleted by the next save.

That means the new dashboard/board behavior only works in injected component/unit tests. With real Firestore data, recorded refunds/carry-forwards are invisible to the dashboard and settlement board, so a refunded household can still appear overpaid and the Owed-to-Members KPI will not reflect resolved credit state. Please thread `creditAdjustments` through normalization, service state, save payload, reset/default paths, and add a service or persistence test proving a loaded year with `creditAdjustments` reaches `useBillingData()` consumers and is preserved on save.

Two smaller follow-ups from the same review pass are worth handling with that fix:

- `getCreditAdjustmentTotalForMember()` currently subtracts every non-cancelled record in `creditAdjustments`, regardless of `type` (`src/lib/calculations.js:87-90`). Since the calculation owns refund/carry-forward dispositions, filter to those types so future records in the same array cannot silently reduce Net Contribution.
- Expanded linked-member rows still use gross `childPayment` for child status and Balance (`src/app/components/SettlementBoard.jsx:331-361`). If child rows remain visible, they should either use the same net/epsilon treatment for that member or drop child-level status/balance so the expanded view cannot disagree with the household-grain truth.

File: src/app/views/Dashboard/DashboardView.jsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ import ConfirmDialog from '../../components/ConfirmDialog.jsx';
  * Port of renderDashboardStatus() from main.js.
  */
 export default function DashboardView() {
-    const { activeYear, familyMembers, bills, payments, loading, service } = useBillingData();
+    const { activeYear, familyMembers, bills, payments, creditAdjustments = [], loading, service } = useBillingData();
```
[the remark this answers is not in the document]
Comment (nathanjohnpayne) [reply src/app/views/Dashboard/DashboardView.jsx:22]: Fixed in 7090948. `creditAdjustments` is now loaded into service state (`normalizeYearData` extracts it, `_loadYearData` stores it) and preserved by `buildSavePayload`, so `DashboardView`/`SettlementBoard` see real records instead of always `[]`, and a save no longer drops them (`setDoc` without merge). Covered by service-level tests: "loads creditAdjustments from the year document into state" and "preserves creditAdjustments in the save payload".

File: src/lib/calculations.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply src/lib/calculations.js:186]: [mergepath-resolve: addressed-elsewhere] Fixed in 0843b96 — `totalOutstanding` is now a sum of per-household net shortfalls and `percentage` derives from it, so a refund or overpayment no longer masks another household's debt. Covered by new tests.

File: src/app/views/Dashboard/DashboardView.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply src/app/views/Dashboard/DashboardView.jsx:22]: [mergepath-resolve: addressed-elsewhere] Fixed in 7090948 — `creditAdjustments` is loaded into service state and preserved by `buildSavePayload` (lossless load→state→save round-trip), so the view no longer reads an always-empty array and a save no longer drops records. Covered by new tests.

File: src/lib/persistence.js [hunk 4]
```diff
@@ -36,6 +38,7 @@ export function buildSavePayload(currentBillingYear, familyMembers, bills, payme
         familyMembers: familyMembers,
         bills: bills,
         payments: payments,
+        creditAdjustments: creditAdjustments,
```
[the remark this answers is not in the document]
Comment (nathanjohnpayne) [reply src/lib/persistence.js:41]: Fixed in 5728fe5. The legacy `/site/` app now round-trips `creditAdjustments` too: `loadBillingYearData` loads `normalized.creditAdjustments` (and resets it on the missing-year branch to avoid cross-year contamination), and `saveData` passes it as the 7th arg to `buildSavePayload`, so the legacy full-document `set()` no longer erases disposition records on the shared document. New legacy test *"writes creditAdjustments in the legacy save payload"* asserts it. Consumer parity restored — both apps round-trip losslessly.

[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply src/lib/persistence.js:41]: [mergepath-resolve: rebuttal-recorded] agent rebuttal posted on thread; resolving.

File: src/main.js [hunk 5]
```diff
@@ -483,6 +484,7 @@ async function loadBillingYearData(yearId) {
         familyMembers = normalized.members;
         bills = normalized.bills;
         payments = normalized.payments;
+        creditAdjustments = normalized.creditAdjustments;
```
[the remark this answers is not in the document]
Comment (nathanjohnpayne) [reply src/main.js:487]: Fixed in 88cefa1, and I swept all remaining consumers so this converges rather than surfacing one more next round:

- `calculateOutstandingBalance` (billing-year.js, the legacy close-message path) now sums per-household **net** shortfalls via `getHouseholdFinancials`, mirroring `calculateSettlementMetrics`; it takes an optional `creditAdjustments` arg.
- The legacy `calculateSettlementMetrics()` wrapper and `confirmCloseYear` now pass the loaded `creditAdjustments`.

Every settlement-math consumer now threads `creditAdjustments`: React metrics + board, React/legacy persistence (load + save), and now legacy metrics + close. I grepped the call sites to confirm none are left on gross. For valid #316 data (no dispositions exist yet) results are unchanged; this aligns the legacy reference app with React for future refund/carry-forward data. New tests: React *"calculateOutstandingBalance … surfaces a shortfall"* and legacy *"incorporates creditAdjustments via the wrapper"*.

[the remark this answers is not in the document]
Comment (nathanpayne-claude) [reply src/main.js:487]: [mergepath-resolve: rebuttal-recorded] agent rebuttal posted on thread; resolving.

**Review by nathanpayne-codex — APPROVED** [review]
Round 4 re-review approved at `88cefa1202a30d835dbe7b0e8cdf8f6ac44957b3`.

The prior blocking persistence issue is resolved: `creditAdjustments` now round-trip through shared persistence, React `BillingYearService` load/save state, and the legacy `/site/` full-document save path, so an ordinary save no longer drops disposition records.

The settlement math is now consistently net-aware across dashboard metrics, settlement board balance/status/action gating, legacy wrapper metrics, and close-year outstanding balance. The over-disposition case degrades to an honest collectable shortfall instead of splitting status from balance.

Verification:
- `npm test` passed locally: legacy tests, React/Vitest tests, and tracked secret scan.
- GitHub Actions current-head `test` passed, including E2E and coverage.
- GitHub Actions current-head `lint`/repo-lint passed.
- `scripts/resolve-pr-threads.sh 323 --repo nathanjohnpayne/friends-and-family-billing --list` reports no unresolved threads.

No remaining blocking findings from this review.
